docker/user.sh

```diff
@@ -52,6 +52,28 @@ then
     mkdir --parents "$HOME"/output
 fi
 
+# Prepare the python virtual environment. We copy the mounted directory if it exists to `.python_venv` to fix the symbolic
+# links to the Python interpreter in the container without affecting the files on host.
+if [[ -d "$HOME/python_venv" ]];
+then
+    cp -r "$HOME/python_venv" "$HOME/.python_venv"
```

Comment:
I wonder if it would be a good idea to name these two `.python_venv` and `python_venv` differently?
It's not quite clear to me at the moment which of it is read-only (i.e. provided by the user) and read-write-able (the one we copied to `.python_venv`) from reading this file alone.
For example:
- `python_venv` -> `input_python_venv`: The virtual env mounted from the host file system
- `.python_venv` -> `deps_python_venv`: The write-able virtual env that only exists in the container.